restli-server/src/main/java/com/linkedin/restli/internal/server/response/CollectionResponseBuilder.java

```diff
@@ -153,8 +154,13 @@ private D buildRestLiResponseData(final Request request,
         throw new RestLiServiceException(HttpStatus.S_500_INTERNAL_SERVER_ERROR,
             "Unexpected null encountered. Null element inside of a List returned by the resource method: " + routingResult.getResourceMethod());
       }
+      DataMap rawData = entry.data();
+      if (resourceContext.isFillInDefaultsRequested())
+      {
+        rawData = ResponseUtils.fillInDefaultValues(entry.schema(), rawData);
+      }
```

Comment:
did you look at any other possible position where this can be inserted instead of response builders? and if not how did we ensure all cases are(will remain) covered where the response is being sent out

Reply:
The way I find the response builder is by looking up all classes that implements the interface `RestLiResponseBuilder` so there are only 10 classes, I only need to implement this default filling feature in classes : `ActionResponseBuilder` `BatchFinderResponseBuilder` `BatchGetResponseBuilder` `GetResponseBuilder` and `CollectionResponseBuilder`, it covers: Action, Get, GetAll, BatchGet, Finder, BatchFinder methods

There are other classes that implements this inferface `RestLiResponseBuilder` I think they don't need to have this:
`BatchCreateResponseBuilder`
`BatchResponseBuilder` for  BATCH_UPDATE, BATCH_PARTIAL_UPDATE
`CreateResponseBuilder`
`EmptyResponseBuilder`
`PartialUpdateResponseBuilder`

Let me know if I miss anything